content/index.md

```diff
@@ -0,0 +1,172 @@
+
+### Action Commands
+
+Action commands return any output that is needed, even if it is just "OK". They also return an appropriate exit status.
+
+All non-data-returning commands support a `--quiet` switch to suppress output.
+
+### Data Comands
+
+Any command that returns data should do so in the following ways.
+
+**Single value**
+
+A command that returns a single value can simply output that value.
+
+**Multiple values**
+
+Even for a single row, this data should be returned in columnar format, with headings. Example:
+
+```
+msisdn       | country | type       | features  | voiceCallbackType | voiceCallbackValue                   | moHttpUrl                                        | voiceStatusCallbackUrl
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+442038795229 | GB      | landline   | VOICE     | app               | 3050a697-f298-4601-8476-7805eafd33fc | undefined                                        | undefined             
+442038931433 | GB      | landline   | VOICE     | app               | 048f0fb1-39a8-40c4-1149-6a2260771fa1 | undefined                                        | undefined             
+442038931435 | GB      | landline   | VOICE     | app               | 048f0fb1-39a8-40c4-9d11-6a2260771fa1 | undefined                                        | undefined             
+447418340330 | GB      | mobile-lvn | VOICE,SMS | undefined         | undefined                            | undefined                                        | undefined             
+447418343340 | GB      | mobile-lvn | VOICE,SMS | undefined         | undefined                            | https://example.ngrok.io/webhooks/inbound-sms | undefined             
+447418343838 | GB      | mobile-lvn | VOICE,SMS | app               | a7fb2d02-92d6-43b4-ae58-6011eae58bbc | https://example.ngrok.io/incoming/sms         | undefined             
+```
+
+This standard approach lends itself well to use with common commands such as `cut` and `tail` that are in common use by developers.
+
+**JSON Format**
+
+For commands that return multiple values, adding a `--format-json` switch will return a pretty-printed JSON response instead of the standard data output.
+
+**Pagination**
+
+Especially for "power users" the size of the data responses can be large. All commands that return multiple values support common pagination parameters:
+* `--limit` The number of records to return. Default: 100
+* `--offset` How many records to skip before returning data. Default: 0
+
+## Error Handling
+
+When an error occurs, either in the CLI or in the underlying API response, the tool needs to respond in a predictable manner, including:
+
+* return an error exit status
+* show the detail of any error from the API
+* add any additional explanation, links or other references to explain to the user what went wrong or what they should do next
```

Comment:
this would be a pain to maintain, we'd need a hash map of errors and our custom addition to the errors.

Reply:
We should only rely on the information in the API for now imo. In some cases (e.g. VAPI in the PHP lib) we get data in the response that we don't return to the user which is useful.

In the future we may choose to enhance these messages in the CLI with our own data should we be unable to update the API itself

Reply:
fair enough, I thought this might be a valuable way to standardise any additional tips we wanted to add for the CLI context (which will be rare but probably not unheard of). We can specifically say that no further hints or feedback should be provided though

Reply:
Removed.